src/r_main.c

```diff
@@ -428,6 +428,63 @@ angle_t R_PointToAngle2(fixed_t pviewx, fixed_t pviewy, fixed_t x, fixed_t y)
 		0;
 }
 
+angle_t R_PointToAngleEx(INT32 x2, INT32 y2, INT32 x1, INT32 y1)
+{
+	INT64 dx = x1-x2;
+	INT64 dy = y1-y2;
+	if (dx < INT32_MIN || dx > INT32_MAX || dy < INT32_MIN || dy > INT32_MAX)
+	{
+		x1 = (int)(dx / 2 + x2);
+		y1 = (int)(dy / 2 + y2);
+	}
+	return (y1 -= y2, (x1 -= x2) || y1) ?
+	x1 >= 0 ?
+	y1 >= 0 ?
+		(x1 > y1) ? tantoangle[SlopeDivEx(y1,x1)] :                            // octant 0
+		ANGLE_90-tantoangle[SlopeDivEx(x1,y1)] :                               // octant 1
+		x1 > (y1 = -y1) ? 0-tantoangle[SlopeDivEx(y1,x1)] :                    // octant 8
+		ANGLE_270+tantoangle[SlopeDivEx(x1,y1)] :                              // octant 7
+		y1 >= 0 ? (x1 = -x1) > y1 ? ANGLE_180-tantoangle[SlopeDivEx(y1,x1)] :  // octant 3
+		ANGLE_90 + tantoangle[SlopeDivEx(x1,y1)] :                             // octant 2
+		(x1 = -x1) > (y1 = -y1) ? ANGLE_180+tantoangle[SlopeDivEx(y1,x1)] :    // octant 4
+		ANGLE_270-tantoangle[SlopeDivEx(x1,y1)] :                              // octant 5
+		0;
+}
+
+//
+// R_PointToPseudoAngle
+// ! Returns the pseudoangle between the line p1 to (infinity, p1.y) and the
+// line from p1 to p2. The pseudoangle has the property that the ordering of
+// points by true angle anround p1 and ordering of points by pseudoangle are the
+// same.
+//
+// For clipping exact angles are not needed. Only the ordering matters.
+// This is about as fast as the fixed point R_PointToAngle2 but without
+// the precision issues associated with that function.
+//
+
+angle_t R_PointToPseudoAngle(fixed_t x, fixed_t y)
+{
+	// Note: float won't work here as it's less precise than the BAM values being passed as parameters
+	double vecx = (double)x - viewx;
+	double vecy = (double)y - viewy;
+	double epsilon = 1e-9;
+
+	if (fabs(vecx) < epsilon && fabs(vecy) < epsilon)
```

Comment:
well im just a little unsure, it used to check if both vec are zero, which the compiler didnt like, but comparing if its smaller than 1e-9 epsilon should be fine?

Reply:
Compiler complains because even though double is pretty precise, it is still not perfect so two numbers that are logically same might give you result that is very close to zero but not equals zero, so "double equals zero" will be false almost always. Tho here you can just change type from double to INT64 and check if it equals 0 (it will be faster and still correct since both y and viewy are fixed_t, which is integer)

Didn't pay attention to this part at first, my bad

Reply:
oki changed it to INT64 
![1186294931891953768](https://github.com/Indev450/SRB2Kart-Saturn/assets/136046486/9e31f7b9-bd04-42d8-864c-3b0406c77a9a)

Reply:
lmao nvm it broke everything